ETHCHAT.md

```diff
@@ -0,0 +1,178 @@
+-   The AD output from X3DH becomes the AD input of the double ratchet. See sections 3.4 and 3.5 of  [https://signal.org/docs/specifications/doubleratchet/](https://signal.org/docs/specifications/doubleratchet/)  for a detailed description.
+-   Bob’s signed prekey SigSPKB from X3DH is used as Bob’s initial ratchet public key of the double ratchet.
+
+Once this initialization has been set, Alice and Bob can start exchanging messages with forward secrecy and authentication.
+
+## Considerations with regards to the Noise protocol
+
+X3DH has three phases:
+
+1.  Bob publishes his identity key and prekeys to a server, or dedicated smart contract.
+2.  Alice fetches a "prekey bundle" from the server, and uses it to send an initial message to Bob.
+3.  Bob receives and processes Alice's initial message.
+
+One observes that, at the beginning of the protocol, the receiver gets the public key through a server, a smart contract in our situation, together with an encrypted ephemeral key. 
+This corresponds to the Noise pattern  **IX**:
+
+→ e, s \
+← e, s, es, se, ee
+
+Concerning the double ratchet, one observes that the Diffie-Hellman ratchet begins with the first message,
+which includes the public key of the sender therefore one may consider the DH ratchet as a  **NN** pattern according to the Noise nomenclature:
+
+→ e \
+← e, ee
+
+When it comes to the KDF ratchet, and the associated encryption process does not correspond exactly to any particular Noise pattern, but one observes that KDF and encryption/decryption functions considered by the double ratchet protocol are included in Noise: 
+SHA256 for the KDF and AES256 for AEAD encryption.
+
+Therefore one may consider the double ratchet algorithm a sort of combination between the Noise protocol, which can be applied in the initialization phase and for the DH ratchet.
+
+Therefore, according to Noise specification, the X3DH algorithm is encoded as  **Noise_IX_25519_AES256GCM_SHA256**
+
+# Retrieving information
+
+## Static data
+
+Some data, such as the key pairs (ik, IK) for Alice and Bob, do not need to be regenerated after a period of time. 
+Therefore the public keys IK can be stored in long-term storage solutions, such as a dedicated smart contract which outputs such a key pair when receiving an Ethereum wallet address.
+
+## Ephemeral data
+
+On the other hand, there is data which is ephemeral and needs to be replaced recurringly, namely Alice’s (ek, EK) or Bob’s (spk, SPK). 
+In this case one could explore storing such data in systems like Waku, as suggested in  [https://rfc.vac.dev/spec/20/](https://rfc.vac.dev/spec/20/). 
+The fact of Bob not updating the ephemeral information can be understood as Bob not willing to participate in any communication.
+
+# Extension to group chat
+
+## 1-to-1 version
+
+In order to extend the protocol to a group chat, one uses an ADKG to replace the X3DH step in the previous combination X3DH + Double Ratchet.
```

Comment:
since this is the first reference of ADKG, maybe an expansion would be good here!